### PR TITLE
Ruby Capstone Project

### PR DESCRIPTION
**Pull request by henask12**
This is the PR for the [Ruby Capstone Project](https://github.com/microverseinc/curriculum-ruby/blob/main/group-capstone/catalog_of_my_things.md) of the Ruby curriculum at [Microverse](https://www.microverse.org/). The goal of this project is to build a command-line application that allows users to manage their collections of things. The application will allow users to create and read items in their collections.

### Command-line Application Goals:

**The** goals of this project are to:**

- [x] Create a command-line application that allows users to manage their collections of things that they own; books, music albums and games.
- [x] Allow users to create and read items in their collections.
- [x]  List Items: Users can retrieve a comprehensive list of all items in their collection, categorized by type (books, music albums, movies, and games).
- [x] Add Items: Users have the capability to add new items to their collection. Whether it's a new book, music album or game, users can input details about their acquisitions.
- [x] Archive Items: The application offers an archiving feature that automatically determines if an item should be archived based on specific criteria (e.g., published date, condition, last played date).
- [x] Store the data in JSON files. All data entered by users is stored persistently in JSON files, ensuring that their collections remain intact even after exiting the application.

**Project Goals:**
 
- [x] Use a kanban board to manage project tasks and allocate work efficiently among team members. Ensure that all project requirements are appropriately distributed and completed.
- [x]  Develop a database schema that mirrors the structure of the application's class hierarchy.
- [x]  Follow the [GitFlow as a workflow](https://github.com/microverseinc/curriculum-transversal-skills/blob/main/git-github/articles/gitflow.md) to create and unify the different features of the project.
- [x] Adhere to established [best practices for Ruby programming](https://github.com/microverseinc/curriculum-ruby/blob/main/articles/ruby_best_practices.md), ensuring code quality, readability, and maintainability.
- [x] Documented the work in a [professional way](https://github.com/microverseinc/curriculum-transversal-skills/blob/main/documentation/articles/professional_repo_rules.md), including code comments and a video presentation. The video presentation should highlight the project's features and demonstrate its functionality.
- [x] Implement unit tests for all methods to verify the correctness of the codebase and promote code reliability.
- [x] Peer code reviews were performed for each of the delivered tasks, writing clear, professional, and respectful review comments for team members.

### **Collaborators** 

1. @henask12 
2. @Kidd254 
3. @sergio-usma 
4. @Javivy 

> Thank you for the Review

### CODE REVIEW

**Review by sergio-usma — DISMISSED** [review]
Ok guys, we are ready to the CR from Microverse!

File: classes/item.rb [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+  def add_author=(author)
+    @author = author
+    author.items.push(self) unless author.items.include?(self)
+  end
+
+  def add_genre=(genre)
+    @genre = genre
+    genre.items.push(self) unless genre.items.include?(self)
+  end
+
```
Comment (algerina) [inline classes/item.rb:35]: - [ ] Well done working on your item class however I recommend adding the source as you are a group of four:


`def source=(source)
      @source = source
      source.items.push(self) unless source.items.include?(self)
    end
`

File: schema.sql [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+);
+
+
+CREATE TABLE Sources (
+    id INT PRIMARY KEY AUTO_INCREMENT,
+    name VARCHAR(255),
+    url VARCHAR(255),
+    item_id INT,
+    FOREIGN KEY (item_id) REFERENCES Item(id)
+);
```
Comment (algerina) [inline schema.sql:62]: - [ ] When testing the app everything is working well but when tried to list all the sources I have get the following error, kindly fix the issue:

Screenshot 2023-09-23 at 4 30 58 PM

**Review by algerina — CHANGES_REQUESTED** [review]
Hi team,

Great job so far!  You just have a small issue to work on.

May I highlight the following: 
-  Well setting up all the classes and methods needed✔
- The code is clean and well organized✔️



## CHANGES REQUIRED:muscle::clap:


Check the comments under the review.



 
 ## Optional suggestions

 - Please check the comment under the review.

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._
Cheers and Happy coding! :clap: :clap: :clap:

Feel free to leave any questions or comments in the PR thread and tag me @algerina if something is not 100% clear.

#### _**Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise.**_
------------------------------------------------------------------------------------------------------------------------------
As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using this [form.](https://airtable.com/shrQAqnBwek5a0O0s)
___
_If anything seems to be unclear to you, reach out and ask anything by commenting on this Pull Request and mention me as [Algerina](https://github.com/algerina), I will receive a notification and will respond._

[GitHub](https://github.com/algerina) | [LinkedIn](https://linkedin.com/in/amel-khiri) | [Twitter](https://twitter.com/AmelQahwadji)

_Reviewed by [Amel Khiri ](https://github.com/algerina)_ ,

Great job so far!  You just have a small issue to work on.

May I highlight the following: 
-  Well done setting up associations between model according to the ERD diagram✔
- The code is clean and well organized✔️



## CHANGES REQUIRED:muscle::clap:


Check the comments under the review.



 
 ## Optional suggestions

 - Please check the comment under the review.

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._
Cheers and Happy coding! :clap: :clap: :clap:

Feel free to leave any questions or comments in the PR thread and tag me @algerina if something is not 100% clear.

#### _**Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise.**_
------------------------------------------------------------------------------------------------------------------------------
As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using this [form.](https://airtable.com/shrQAqnBwek5a0O0s)
___
_If anything seems to be unclear to you, reach out and ask anything by commenting on this Pull Request and mention me as [Algerina](https://github.com/algerina), I will receive a notification and will respond._

[GitHub](https://github.com/algerina) | [LinkedIn](https://linkedin.com/in/amel-khiri) | [Twitter](https://twitter.com/AmelQahwadji)

_Reviewed by [Amel Khiri ](https://github.com/algerina)_

File: classes/item.rb (hunk 1 above)
Comment (rloterh) [reply classes/item.rb:35]: well resolved! ✔️

File: schema.sql (hunk 2 above)
Comment (rloterh) [reply schema.sql:62]: well resolved! ✔️

**Review by rloterh — APPROVED** [review]
## Status: Approved :heavy_check_mark:

Hi Team,  

**Ruby group capstone** project is complete! 🏁 Great job! Congratulations!  🎉💯🌟 

## Highlights 🎯 
🎲 Good job improving on the app functionality!

Cheers..Happy coding!..and keep soaring higher! 💻 🍷 🚀 

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. Remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think the code review was unfair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**